engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/package.scala

```diff
@@ -9,23 +9,62 @@ package object callcaching {
 
-  sealed trait CallCachingMode { def activity: Option[CallCachingActivity]; def readFromCache = false; def writeToCache = false; def lookupDockerHashes: Boolean = false }
+  sealed trait CallCachingMode {
+    def activity: Option[CallCachingActivity]
+
+    def readFromCache = false
+    def writeToCache = false
+    def hashDockerNames = false
+    def lookupDockerHashes: Boolean = false
+    def hashFilePaths: Boolean = false
+    def hashFileContents: Boolean = false
+
+    /**
+      * Return an equivalent of this call caching mode with READ disabled.
+      */
+    def withoutRead: CallCachingMode
+  }
+
   object CallCachingMode {
-    def apply(read: Boolean, write: Boolean, lookupDockerHashes: Boolean): CallCachingMode = (read, write) match {
+    def apply(read: Boolean, write: Boolean, hashDockerNames: Boolean, lookupDockerHashes: Boolean, hashFilePaths: Boolean, hashFileContents: Boolean): CallCachingMode = (read, write) match {
       case (false, false) => CallCachingOff
-      case (false, true) => WriteCache(lookupDockerHashes)
-      case (true, false) => ReadCache(lookupDockerHashes)
-      case (true, true) => ReadAndWriteCache(lookupDockerHashes)
+      case (false, true) => WriteCache(hashDockerNames, lookupDockerHashes, hashFilePaths, hashFileContents)
+      case (true, false) => ReadCache(hashDockerNames, lookupDockerHashes, hashFilePaths, hashFileContents)
+      case (true, true) => ReadAndWriteCache(hashDockerNames, lookupDockerHashes, hashFilePaths, hashFileContents)
     }
   }
 
-  case object CallCachingOff extends CallCachingMode { override def activity = None; }
+  case object CallCachingOff extends CallCachingMode {
+    override def activity = None
+    override def withoutRead = this
+  }
 
-  sealed trait CallCachingActivity extends CallCachingMode { override def activity = Option(this); }
-  case class ReadCache(override val lookupDockerHashes: Boolean) extends CallCachingActivity { override def readFromCache = true }
-  case class WriteCache(override val lookupDockerHashes: Boolean) extends CallCachingActivity { override def writeToCache = true }
-  case class ReadAndWriteCache(override val lookupDockerHashes: Boolean) extends CallCachingActivity { override def readFromCache = true; override def writeToCache = true }
+  sealed trait CallCachingActivity extends CallCachingMode {
+    override def activity = Option(this)
+  }
+  case class ReadCache(override val hashDockerNames: Boolean, override val lookupDockerHashes: Boolean, override val hashFilePaths: Boolean, override val hashFileContents: Boolean) extends CallCachingActivity {
+    override def readFromCache = true
+    override def withoutRead = CallCachingOff
+  }
+  case class WriteCache(override val hashDockerNames: Boolean, override val lookupDockerHashes: Boolean, override val hashFilePaths: Boolean, override val hashFileContents: Boolean) extends CallCachingActivity {
```

Comment:
you shouldn't need the `override val` stuff here. 

Also these case classes should be final